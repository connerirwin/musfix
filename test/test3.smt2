(qualif Zog ((v a)(z b)) (= v z))

(wf $k0 ((v int)))

(constraint
  (forall ((v int)(q int))
          (=> (= v q) ($k0 q))))

(constraint
  (forall ((v int))
          (=> ($k0 10) (= v 10))))
