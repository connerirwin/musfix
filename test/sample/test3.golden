
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/sample/test3.smt2
Qualifier "Zog" [Var (VarS "a") "v",Var (VarS "b") "z"] (Binary Eq (Var (VarS "a") "v") (Var (VarS "b") "z"))
WFConstraint "$k0" [Var IntS "v",Var IntS "x"]
HornConstraint [Var IntS "v",Var IntS "q"] (Binary Implies (Binary Eq (Var IntS "v") (Var IntS "q")) (Unknown (fromList [("v",Var IntS "v"),("x",Var IntS "q")]) "$k0"))
HornConstraint [Var IntS "v",Var IntS "y"] (Binary Implies (Binary And (Unknown (fromList [("v",Var IntS "v"),("x",Var IntS "y")]) "$k0") (Binary Eq (Var IntS "y") (IntLit 10))) (Binary Eq (Var IntS "v") (IntLit 10)))

QMAP
--------
fromList [("$k0",QSpace {_qualifiers = [Binary Eq (Var IntS "v") (Var IntS "x"),Binary Eq (Var IntS "x") (Var IntS "v")], _maxCount = 2})]

Candidates
--------
[Candidate {solution = fromList [("$k0",fromList [Binary Eq (Var IntS "x") (Var IntS "v")])], validConstraints = fromList [Binary Implies (Binary Eq (Var IntS "v") (Var IntS "q")) (Unknown (fromList [("v",Var IntS "v"),("x",Var IntS "q")]) "$k0"),Binary Implies (Binary And (Unknown (fromList [("v",Var IntS "v"),("x",Var IntS "y")]) "$k0") (Binary Eq (Var IntS "y") (IntLit 10))) (Binary Eq (Var IntS "v") (IntLit 10))], invalidConstraints = fromList [], label = "0.0"},Candidate {solution = fromList [("$k0",fromList [Binary Eq (Var IntS "v") (Var IntS "x")])], validConstraints = fromList [Binary Implies (Binary Eq (Var IntS "v") (Var IntS "q")) (Unknown (fromList [("v",Var IntS "v"),("x",Var IntS "q")]) "$k0"),Binary Implies (Binary And (Unknown (fromList [("v",Var IntS "v"),("x",Var IntS "y")]) "$k0") (Binary Eq (Var IntS "y") (IntLit 10))) (Binary Eq (Var IntS "v") (IntLit 10))], invalidConstraints = fromList [], label = "0.1"}]


Final candidates: 
0.0: [$k0 -> {x [37m==[0m v}] (2 0)
0.1: [$k0 -> {v [37m==[0m x}] (2 0)

